tests/console_test.go

```diff
@@ -55,6 +56,7 @@ var _ = Describe("[rfe_id:127][posneg:negative][crit:medium][vendor:cnv-qe@redha
 
 	RunVMIAndWaitForStart := func(vmi *v1.VirtualMachineInstance) {
 		By("Creating a new VirtualMachineInstance")
+		vmi.ObjectMeta.Namespace = util.NamespaceTestDefault
```

Comment:
Same

Reply:
Ohh my, after looking deeper I see why you needed this.

First, lets ignore that the creation of the VMI is done using the Rest client (sad).
On creation, the VMI needs to be updated with the latest values, which includes the namespace.

A quick (ugly) fix will be to add this at the end of this helper:
```
updatedVMI, err := virtClient.VirtualMachineInstance(util.NamespaceTestDefault).Get(vmi.Name, &metav1.GetOptions{})
Expect(err).ToNot(HaveOccurred())
*vmi = *updatedVMI
```

A better fix will be to return here the `vmi` created to the caller, so it can be updated.
And if we really want to leave it in a sane state, then just use the proper creation version:
`vmi, err := virtClient.VirtualMachineInstance(util.NamespaceTestDefault).Create(vmi)`
(the return `vmi` includes all the data upon creation, including the namespace)

Reply:
> A better fix will be to return here the vmi created to the caller, so it can be updated.

I don't understand your gist. `RunVMIAndWaitForStart` needs to get the vmi started, not to return it for further updates by the caller. I believe that if I do not set ObjectMeta.Namespace prior to it, `vmi, err := virtClient.VirtualMachineInstance(util.NamespaceTestDefault).Create(vmi)` returns the same `an empty namespace may not be set when a resource name is provided` error. I'll take a deeper look.

Reply:
`FUNC_TEST_ARGS='-focus=console_test -regexScansFilePath' make functest` now passes locally, let's see what CI thinkgs of it.